cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/LoginAction.java

```diff
@@ -121,9 +123,13 @@ public void doAction() throws Exception {
             throw new AccessControlException("login failed");
         }
         
-        // TODO Alinga
-        // Add group check on rp.accessGroup here
-        // (will require client_id to be passed from AuthorizeAction, to oidc-login.html, to here)
+        // perform group check on rp.accessGroup 
+        // (use clientID passed from AuthorizeAction, to oidc-login.html)
+        if (!OIDCUtil.accessAllowed(clientID)) {
```

Comment:
A check for null client_id has to be done (as is done with redirect_uri, username, password)

Reply:
Also, the check for a RelyParty with the clientID needs to be done here.  The rely party should be the param sent to OIDCUtil.accessAllowed()

Reply:
Good catch.